src/main/java/com/zaxxer/nuprocess/NuProcessBuilder.java

```diff
@@ -252,17 +252,42 @@ public void setCwd(Path cwd)
     *         immediately detectable launch failure
     */
    public NuProcess start()
+   {
+      ensureListener();
+      String[] env = prepareEnvironment();
+
+      return factory.createProcess(command, env, processListener, cwd);
+   }
+
+   /**
+    * Spawn the child process with the configured commands, environment, and {@link NuProcessHandler}
+    * and wait for it to complete running.
+    *
+    * @since 1.3
+    */
+   public void run()
```

Comment:
Me and @bturner have considered to return several things here:

- The `NuProcess` itself. This doesn't make sense as what for would you use it, given that the method returns after process is exited?
- The status code from syscalls that start processes. I struggle to imagine use case for this one.
- The exit code of the process. But in this case it would duplicate the code that is sent to `NuProcessHandler#onExit`.

We've chosen to return nothing but we really don't have strong opinion on that matter and ready to change it to your preferred way if you have one.

The same comment applies for whole chain of `run` methods in builder and factory.